blobs.go

```diff
@@ -75,6 +75,17 @@ type BlobDeleter interface {
 	Delete(ctx context.Context, dgst digest.Digest) error
 }
 
+// BlobEnumerator allows to list blobs in storage.
+type BlobEnumerator interface {
+	// Enumerate fills 'digests' with a lexicographically sorted list of
+	// digests up to the size of 'digests' and returns the value 'n' for the
+	// number of entries which were filled. 'last' contains an offset in the
+	// list. It's a string representation of digest. Empty string means 0
+	// offset. 'err' will be set to io.OEF if there are no more entries to
+	// obtain.
+	Enumerate(ctx context.Context, digests []digest.Digest, last string) (n int, err error)
```

Comment:
@stevvooe No problem. I'm just not sure, what do you mean by a callback approach. Few ideas come to my mind.

1. I fear this would make clients nontrivial.

    // Enumerate calls `DigestIngester()` in a loop with a # of digests up to
    // the batchSize until the ingester returns `false` or entire blob store is
    // processed. The ingester will be called with `io.EOF` when no more
    // digests are available.
    Enumerate(ctx context.Context, batchSize int, ingester DigestIngester) error

    type DigestIngester func(digests []digest.Digest, err error) (continue bool)

2. Simple but not that efficient.

    // Enumerate returns an iterator which, upon invocation, produces a single
    // digest until an entire blob store is processed. In that case, empty
    // digest and `io.EOF` will be returned.
    Enumerate(ctx context.Context) (iter DigestIterator, err error)

    type DigestIterator func() (digest.Digest, err error)

3. My favorite.

    // Enumerate returns an enumerator function that can be iteratively called
    // by client with a non-zero long digest array that will be filled. When
    // all the digests are processed, io.EOF will be returned.
    Enumerate(ctx context.Context) (enumerator DigestEnumerator, err error)

    type DigestEnumerator func([]digest.Digest) (err error)

Did you have anything else on your mind?